### PR TITLE
add parameter c

### LINKED ISSUE #3: Parameter -c

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #3

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -32,6 +41,11 @@ func main() {
 			inputAsInt[i][sid] = value
 		}
 
+		if *fieldsParameter != "" && *charParameter != "" {
+			fmt.Println("cut: only one type of list may be specified.")
+			os.Exit(1)
+		}
+
```
Comment (Al2Klimov) [inline main.go]: Ensure this one even before computing chosenParameter.

File: main.go [hunk 2]
```diff
@@ -32,6 +41,11 @@ func main() {
 			inputAsInt[i][sid] = value
 		}
 
+		if *fieldsParameter != "" && *charParameter != "" {
+			fmt.Println("cut: only one type of list may be specified.")
```
Comment (Al2Klimov) [inline main.go]: Where are you printing the error message?

File: main.go [hunk 3]
```diff
@@ -32,6 +41,11 @@ func main() {
 			inputAsInt[i][sid] = value
 		}
 
+		if *fieldsParameter != "" && *charParameter != "" {
```
Comment (Al2Klimov) [inline main.go]: Please explain why `==` would be better than `&&`.

File: main.go [hunk 4]
```diff
@@ -66,9 +84,12 @@ func main() {
...
 
-		contentLineSplit[0] = bytes.Join(output, delimiter)
-
-		_, _ = os.Stdout.Write(bytes.Join(contentLineSplit, []byte{'\n'}))
+		if *fieldsParameter != "" {
+			contentLineSplit[0] = bytes.Join(output, delimiter)
+			_, _ = os.Stdout.Write(bytes.Join(contentLineSplit, []byte{'\n'}))
+		} else if *charParameter != "" {
+			fmt.Printf("%s\n", bytes.Join(output, []byte("")))
+		}
```
Comment (Al2Klimov) [inline main.go]: What's this for?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (justinmueller05082000) [reply main.go]: in line 45 obvly

File: main.go (hunk 4 above)
Comment (justinmueller05082000) [reply main.go]: to print characters and not words

File: main.go (hunk 3 above)
Comment (justinmueller05082000) [reply main.go]: i dont understand the question i cannot use `==`?

File: main.go (hunk 2 above)
Comment (Al2Klimov) [reply main.go]: _Sigh_. Where are you printing the error message **to**?

File: main.go (hunk 3 above)
Comment (Al2Klimov) [reply main.go]: `==` also covers neither -f, nor -c being given which you don't want.

File: main.go (hunk 2 above)
Comment (justinmueller05082000) [reply main.go]: to the terminal

File: main.go (hunk 4 above)
Comment (Al2Klimov) [reply main.go]: Do you actually need this change?

Comment (justinmueller05082000) [reply main.go]: well its not working with just stdout.write and its the qucikest solution ive found

File: main.go (hunk 3 above)
Comment (justinmueller05082000) [reply main.go]: still dont understand it

File: main.go (hunk 2 above)
Comment (Al2Klimov) [reply main.go]: Please open fmt.Println() and answer a bit more precise.

Comment (justinmueller05082000) [reply main.go]: it writes to standard output

File: main.go (hunk 4 above)
Comment (Al2Klimov) [reply main.go]: If I revert this change, it works for me.

File: main.go (hunk 3 above)
Comment (Al2Klimov) [reply main.go]: Shall your program be able to run without -f and -c?

File: main.go (hunk 4 above)
Comment (justinmueller05082000) [reply main.go]: Screenshot from 2020-05-13 12-19-28
it does work but it looks ugly

File: main.go (hunk 2 above)
Comment (Al2Klimov) [reply main.go]: And where do you usually write errors to?

File: main.go (hunk 3 above)
Comment (justinmueller05082000) [reply main.go]: no it cannot be used without parameters

File: main.go (hunk 4 above)
Comment (Al2Klimov) [reply main.go]:
```
➜  trainee-go-cut git:(feature/parameter-c-3) ✗ git diff
diff --git a/main.go b/main.go
index 14ce070..f103d8a 100644
--- a/main.go
+++ b/main.go
@@ -84,12 +84,12 @@ func main() {
                        }
                }

-               if *fieldsParameter != "" {
+               //if *fieldsParameter != "" {
                        contentLineSplit[0] = bytes.Join(output, delimiter)
                        _, _ = os.Stdout.Write(bytes.Join(contentLineSplit, []byte{'\n'}))
-               } else if *charParameter != "" {
+               /*} else if *charParameter != "" {
                        fmt.Printf("%s\n", bytes.Join(output, []byte("")))
-               }
+               }*/

                if readAllErr == io.EOF {
                        break
➜  trainee-go-cut git:(feature/parameter-c-3) ✗ go run . -c 2-3
abcd
bc
➜  trainee-go-cut git:(feature/parameter-c-3) ✗
```

Comment (justinmueller05082000) [reply main.go]: it has something to do with the OS then

File: main.go (hunk 3 above)
Comment (Al2Klimov) [reply main.go]: If you run just `go run .`, both `*fieldsParameter != ""` `*charParameter != ""` are false – and equal. The same for `go run . -f 1 -c 2`, but true – also equal. With `==` you cover both invalid cases.

File: main.go (hunk 4 above)
Comment (Al2Klimov) [reply main.go]: If you make exactly my above changes and nothing more – doesn't it work for you?

Comment (justinmueller05082000) [reply main.go]: if i revert the changes i made it looks like on the picture

Comment (Al2Klimov) [reply main.go]: What does `hexdump -C < toCut.txt` say?

File: main.go [hunk 5]
```diff
@@ -67,7 +85,6 @@ func main() {
 		}
 
 		contentLineSplit[0] = bytes.Join(output, delimiter)
-
```
Comment (Al2Klimov) [inline main.go]: Not part of the task.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 4 above)
Comment (justinmueller05082000) [reply main.go]:
```
00000000  61 09 62 09 63 0a 64 09  65 09 66                 |a.b.c.d.e.f|
0000000b

```

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD feature/add-copying-file-9)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/parameter-c-3`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol.

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**toCut.txt:**
```
abc	def	ghi
jkl	mno	pqr
```

**Go command:**

**Terminal:** `go run . -c 1-5 < toCut.txt`

**Output:**
```
abc	d
jkl	m
```

**Cut command:**

**Terminal:** `cut -c 1-5 < toCut.txt`

**Output:**
```
abc	d
jkl	m
```

**Conclusion:**
Both results are equal.

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**